### PR TITLE
The NIST round 3 Falcon Signature Scheme integration.

### PR DESCRIPTION
**Pull request by anhu**

### CODE REVIEW

File: wolfssl/internal.h [hunk 1]
```diff
@@ -1239,9 +1242,13 @@ enum Misc {
 #else
 #ifndef NO_PSK
     ENCRYPT_LEN     = 512 + MAX_PSK_ID_LEN + 2,    /* 4096 bit static buffer */
+#else
+#ifdef HAVE_LIBOQS
+    ENCRYPT_LEN     = 5000,     /* allow 40000 bit static buffer for falcon */
```
Comment (dgarske) [inline wolfssl/internal.h:1247]: Can you verify the 5000 is the actual max used?

File: wolfssl/internal.h [hunk 2]
```diff
@@ -1448,10 +1455,21 @@ enum Misc {
...
+    FALCON_LEVEL5_SA_MAJOR = 0xFE,
+    FALCON_LEVEL5_SA_MINOR = 0x0E,
+
+
     MIN_RSA_SHA512_PSS_BITS = 512 * 2 + 8 * 8, /* Min key size */
     MIN_RSA_SHA384_PSS_BITS = 384 * 2 + 8 * 8, /* Min key size */
 
-#ifndef NO_RSA
+#if defined(HAVE_LIBOQS)
+    MAX_CERT_VERIFY_SZ = 20000,            /* For Falcon */
```
Comment (dgarske) [inline wolfssl/internal.h:1471]: Also check the max on this... Thanks

**Comment by anhu** [comment]
Looking into it now.

**Review by dgarske — APPROVED** [review]

**Review by dgarske — APPROVED** [review]

**Comment by anhu** [comment]
:tada:

**Pull request merged**